stout/range.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "stout/stream.h"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace stout {
+namespace eventuals {
+
+////////////////////////////////////////////////////////////////////////
+
+struct _Range {
+  template <typename K_, typename Arg_>
+  struct Continuation : public detail::TypeErasedStream {
+    // NOTE: explicit constructor because inheriting 'TypeErasedStream'.
+    Continuation(K_ k, int from, int to, int step)
+      : k_(std::move(k)),
+        from_(from),
+        to_(to),
+        step_(step) {}
+
+    void Start() {
+      k_.Start(*this);
+    }
+
+    template <typename... Args>
+    void Fail(Args&&... args) {
+      k_.Fail(std::forward<Args>(args)...);
+    }
+
+    void Stop() {
+      k_.Stop();
+    }
+
+    void Ended() {
+      k_.Ended();
+    }
+
+    void Next() override {
+      if ((from_ == to_) || (step_ == 0) || (from_ > to_ && step_ > 0) || (from_ < to_ && step_ < 0)) {
+        k_.Ended();
+      } else {
+        int temp = from_;
+        from_ += step_;
+        k_.Body(temp);
+      }
```

Comment:
Instead of having to indent this all in an `else`, consider adding a `return` to the `if` block.